.github/workflows/cnf-test-partner-image.yaml

```diff
@@ -0,0 +1,63 @@
+name: 'Test/push the `cnf-test-partner` image'
+on:
+  push:
+    paths:
+      - 'test-partner/Dockerfile'
+  pull_request:
```

Comment:
The expected behavior here is:

- Run on every [Pull Request](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request). 
This will trigger the `test-cnf-test-partner-image` job that builds and tests the Dockerfile. The _push_ job is skipped, as it applies only to the `main` branch.

and

- Run on every [Push](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push) that contains modifications to the Dockerfile. 
After a PR is merged, the workflow will be run again, this time on the `main` branch. 
If all tests pass, the `push-cnf-test-partner-image` job builds the image from `main` and pushes the image to Quay.io.